tests/test_date_levels_and_comparisons.py

```diff
@@ -0,0 +1,373 @@
+                values={"dob_l": "2000-01-01", "dob_r": "2020-01-01"},
+                expected_gamma_val=0,
+            ),
+            LiteralTestValues(
+                values={"dob_l": "2000-01-01", "dob_r": "2000-01-15"},
+                expected_gamma_val=1,
+            ),
+            LiteralTestValues(
+                values={"dob_l": "2000-ab-cd", "dob_r": "2000-01-28"},
+                expected_gamma_val=-1,
+            ),
+        ],
+    )
+
+    run_tests_with_args(test_spec, db_api)
+
+
+@mark_with_dialects_including("duckdb", pass_dialect=True)
+def test_alternative_date_format(test_helpers, dialect):
+    helper = test_helpers[dialect]
+    db_api = helper.extra_linker_args()["database_api"]
+
+    test_spec = ComparisonTestSpec(
+        cl.AbsoluteDateDifferenceAtThresholds(
+            "dob",
+            thresholds=[3, 2],
+            metrics=["day", "month"],
+            input_is_string=True,
+            datetime_format="%Y/%m/%d",
+        ),
+        tests=[
+            LiteralTestValues(
+                values={"dob_l": "2000/01/01", "dob_r": "2020/01/01"},
+                expected_gamma_val=0,
+            ),
+            LiteralTestValues(
+                values={"dob_l": "2000/01/01", "dob_r": "2000/01/15"},
+                expected_gamma_val=1,
+            ),
+            LiteralTestValues(
+                values={"dob_l": "2000/01/01", "dob_r": "2000/01/02"},
+                expected_gamma_val=2,
+            ),
+            LiteralTestValues(
+                values={"dob_l": "2000/ab/cd", "dob_r": "2000/01/28"},
+                expected_gamma_val=-1,
+            ),
+        ],
+    )
+
+    run_tests_with_args(test_spec, db_api)
+
+
+@mark_with_dialects_excluding("sqlite")
+def test_time_difference_error_logger(dialect):
+    # Differing lengths between thresholds and units
+    with pytest.raises(ValueError):
+        cl.AbsoluteDateDifferenceAtThresholds(
+            "dob",
+            thresholds=[1],
+            metrics=["day", "month", "year", "year"],
+            input_is_string=True,
+        )
+    # Negative threshold
+    with pytest.raises(ValueError):
+        cl.AbsoluteDateDifferenceAtThresholds(
+            "dob", thresholds=[-1], metrics=["day"], input_is_string=True
+        )
+    # Invalid metric
+    with pytest.raises(ValueError):
+        cl.AbsoluteDateDifferenceAtThresholds(
+            "dob", thresholds=[1], metrics=["dy"], input_is_string=True
+        )
+    # Threshold len == 0
+    with pytest.raises(ValueError):
+        cl.AbsoluteDateDifferenceAtThresholds(
+            "dob", thresholds=[], metrics=["dy"], input_is_string=True
+        )
+    # Metric len == 0
+    with pytest.raises(ValueError):
+        cl.AbsoluteDateDifferenceAtThresholds(
+            "dob", thresholds=[1], metrics=[], input_is_string=True
+        )
+
+
+@mark_with_dialects_excluding("sqlite", "postgres")
+def test_date_comparison(test_helpers, dialect):
```

Comment:
these are the new tests, it hasn't detected the file rename correctly so everything here looks new